src/vm/memory/relocatable.zig

```diff
@@ -42,6 +42,54 @@ pub const Relocatable = struct {
         return self.segment_index == other.segment_index and self.offset == other.offset;
     }
 
+    // Determines if this Relocatable is less than another.
+    // # Arguments
+    // - other: The other Relocatable to compare to.
+    // # Returns
+    // `true` if self is less than other, `false` otherwise.
+    pub fn lt(
+        self: Self,
+        other: Self,
+    ) bool {
+        return self.segment_index < other.segment_index and self.offset < other.offset;
+    }
+
+    // Determines if this Relocatable is less than or equal to another.
+    // # Arguments
+    // - other: The other Relocatable to compare to.
+    // # Returns
+    // `true` if self is less than or equal to other, `false` otherwise.
+    pub fn le(
+        self: Self,
+        other: Self,
+    ) bool {
+        return self.lt(other) or self.eq(other);
+    }
+
+    // Determines if this Relocatable is greater than another.
+    // # Arguments
+    // - other: The other Relocatable to compare to.
+    // # Returns
+    // `true` if self is greater than other, `false` otherwise.
+    pub fn gt(
+        self: Self,
+        other: Self,
+    ) bool {
+        return !self.lt(other);
```

Comment:
Here I think it doesn't work because if `self` and `other` are equal, then `!self.lt(other)` is true and yet `self` is not greater than `other`. So I think you either have to do `!self.le(other)` or write the comparison in its entirety `self.segment_index > other.segment_index and self.offset > other.offset` which is more efficient from a performance point of view because the code does not need to run both `self.lt(other)` then `self.eq(other)` and then do the negation in the case of equality.

Reply:
yeah typo on my side, its same as you suggested in the python implementation.